frontend/src/metabase/query_builder/components/DatasetEditor/DatasetQueryEditor.unit.spec.tsx

```diff
@@ -0,0 +1,161 @@
+import { screen } from "@testing-library/react";
+import fetchMock from "fetch-mock";
+import _ from "underscore";
+
+import { setupDatabasesEndpoints } from "__support__/server-mocks";
+import { createMockEntitiesState } from "__support__/store";
+import { renderWithProviders } from "__support__/ui";
+import { Card, Collection } from "metabase-types/api";
+import {
+  createMockCard,
+  createMockNativeDatasetQuery,
+} from "metabase-types/api/mocks";
+import { createSampleDatabase } from "metabase-types/api/mocks/presets";
+import { createMockState } from "metabase-types/store/mocks";
+import { checkNotNull } from "metabase/core/utils/types";
+import { getMetadata } from "metabase/selectors/metadata";
+import NativeQuery from "metabase-lib/queries/NativeQuery";
+
+const { NativeQueryEditor } = jest.requireActual(
+  "metabase/query_builder/components/NativeQueryEditor",
+);
+
+const TEST_DB = createSampleDatabase();
+
+const TEST_NATIVE_CARD = createMockCard({
+  dataset_query: createMockNativeDatasetQuery({
+    database: TEST_DB.id,
+  }),
+});
+
+const ROOT_COLLECTION = {
+  id: "root",
+  name: "Our analytics",
+  can_write: true,
+} as Collection;
+
+interface SetupOpts {
+  card?: Card;
+  height?: number;
+  isActive: boolean;
+  readOnly?: boolean;
+}
+
+const setup = async ({
+  card = TEST_NATIVE_CARD,
+  height = 300,
+  isActive,
+  readOnly = false,
+}: SetupOpts) => {
+  setupDatabasesEndpoints([TEST_DB]);
+
+  const storeInitialState = createMockState({
+    entities: createMockEntitiesState({
+      databases: [createSampleDatabase()],
+      questions: [card],
+    }),
+  });
+  const metadata = getMetadata(storeInitialState);
+  const query = new NativeQuery(
+    checkNotNull(metadata.database(TEST_DB.id)).question(),
+    {
+      type: "native",
+      database: TEST_DB.id,
+      native: {
+        query: "select * from orders",
+        "template-tags": undefined,
+      },
+    },
+  );
+  const question = checkNotNull(metadata.question(card.id));
+
+  const { default: DatasetQueryEditor } = await import(
```

Comment:
Why this is required?

Reply:
`NativeQueryEditor` is [globally mocked](https://github.com/metabase/metabase/blob/9ca4af5/frontend/test/register-visualizations.js#L10) but its actual implementation is needed in this test suite because we need to investigate its children.

We're actually testing `NativeQueryEditor` indirectly by using `DatasetQueryEditor` (which uses `NativeQueryEditor`), so the `NativeQueryEditor` has to be unmocked the moment we import `DatasetQueryEditor`.

Unmocking happens in `beforeEach`, so we can really only import the component during the unit test.

Should the import be at the beginning of this file, the [mock `NativeQueryEditor`](https://github.com/metabase/metabase/blob/9ca4af5b3c9c18956e7c0d545fb029d21e7c452e/frontend/src/metabase/query_builder/components/__mocks__/NativeQueryEditor.jsx) would have been used in tests instead of the actual implementation.